### DISCUSSION TITLE
FormBuilder и его методы fillCast и fields

### DISCUSSION THREAD

**Discussion by Mr-Coquin** [moonshine-software/moonshine#1078] [Q&A]
```
ActionButton::make('Modal')
                                        ->inModal('Modal', fn() => FormBuilder::make($fullUrl)
                                            ->async(asyncEvents: ["fragment-updated-maincard{$element->id}"])
                                            ->fillCast(
                                                $element,
                                                $item->getModelCast()
                                            )
                                            ->fields(
 //                                               $item->getFormFields()
text::make('Title', 'title')
                                            ),
                                            wide:true, closeOutside :true),
```
  Я указал в fields какие поля хочу обновить при редактирования он мне все вывел и сохранил. Но остальные поля которые я в  fields не указал он мне обнулил . Можете подсказать как этого не допустить ? При сохранение поля BelongsTo делает null но title меняет . Если оставить $item->getFormFields() то он выводит все поля и при сохранение проблем никаких не вызывает.

**Comment by lee-to** [comment]
Не совсем понимаю но в целом возможно эта мысль вам поможет
Если у вас в ресурсе один набор полей а в модалке другой то сохранятся будет на основе полей ресурса а не модалки, вам в таком случае нужно динамически менять набор полей ресурса в зависимости от реквеста
